processor/telemetryprocessor/telemetry_processor_test.go

```diff
@@ -0,0 +1,216 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetryprocessor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opencensus.io/stats/view"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/collector/telemetry"
+	"go.opentelemetry.io/collector/internal/data/testdata"
+	"go.opentelemetry.io/collector/obsreport/obsreporttest"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+func TestErrorPath(t *testing.T) {
+	doneFn, err := obsreporttest.SetupRecordedMetricsTest()
+	require.NoError(t, err)
+	defer doneFn()
+
+	views := processor.MetricViews(configtelemetry.LevelDetailed)
+	assert.NoError(t, view.Register(views...))
+	defer view.Unregister(views...)
+
+	mockP := newMockSpanProcessor()
+	creationParams := component.ProcessorCreateParams{Logger: zap.NewNop()}
+	cfg := createDefaultConfig().(*Config)
+	qp := newTelemetryTracesProcessor(creationParams, mockP, cfg)
+	require.NoError(t, qp.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() {
+		assert.NoError(t, qp.Shutdown(context.Background()))
+	})
+
+	mockP.consumeError = errors.New("consume error")
+
+	wantBatches := 10
+	for i := 0; i < wantBatches; i++ {
+		td := testdata.GenerateTraceDataTwoSpansSameResource()
+		require.Error(t, qp.ConsumeTraces(context.Background(), td))
+	}
+
+	obsreporttest.CheckProcessorTracesViews(t, cfg.Name(), int64(0), 20, 0)
+}
+
+func TestTraceHappyPath(t *testing.T) {
+	doneFn, err := obsreporttest.SetupRecordedMetricsTest()
+	require.NoError(t, err)
+	defer doneFn()
+
+	lvl, err := telemetry.GetLevel()
+	require.NoError(t, err)
+	telemetry.SetLevel(configtelemetry.LevelDetailed)
+	defer telemetry.SetLevel(lvl)
```

Comment:
this is ugly as well, and also not thread safe, idk if we care.

Reply:
i'd rather not add a mutex, and if I do that I feel like it should be refactored to get rid of the janky level from string logic on the get()